waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -103,5 +127,71 @@ writeMainWaspFile waspProjectDir (ProjectInfo projectName appName) = IOUtil.writ
           "}"
         ]
 
+createProjectFromTemplate :: Path System Abs (Dir WaspProjectDir) -> ProjectInfo -> Command ()
+createProjectFromTemplate absWaspProjectDir ProjectInfo {_appName = appName, _projectName = projectName, _template = maybeTemplateName} = do
+  cliSendMessageC $ Msg.Start "Creating project from template..."
+
+  dirWhereProjectIsCreated <- getDirWhereProjectIsCreated
+  templatePath <- getPathToRemoteTemplate maybeTemplateName
+
+  let projectDir = projectName
+
+  fetchTemplate templatePath projectDir dirWhereProjectIsCreated
+  ensureTemplateWasFetched
+
+  replacePlaceholdersInWaspFile
+  where
+    getDirWhereProjectIsCreated :: Command (Path System Abs (Dir dir))
+    getDirWhereProjectIsCreated = do
+      (liftIO getCurrentDirectory <&> parseAbsDir) >>= \case
+        Left err -> throwProjectCreationError $ "Failed to parse absolute path to current working directory: " ++ show err
+        Right result -> return result
+
+    getPathToRemoteTemplate :: Maybe String -> Command String
+    getPathToRemoteTemplate = \case
+      Just templateName -> return $ waspTemplatesRepo ++ "/" ++ templateName
+      Nothing -> throwProjectCreationError "Template name is not provided."
+      where
+        -- gh: prefix means Github repo
+        waspTemplatesRepo = "gh:wasp-lang/starters"
+
+    fetchTemplate :: String -> String -> Path System Abs (Dir dir) -> Command ()
+    fetchTemplate templatePath projectDir dirWhereProjectIsCreated = do
+      liftIO executeCmd >>= \case
+        ExitSuccess -> return ()
+        ExitFailure _ -> throwProjectCreationError "Failed to create project from template."
+      where
+        executeCmd = newChan >>= runNodeCommandAsJob dirWhereProjectIsCreated "npx" args J.Cli
```

Comment:
Actually let's not use `runNodeCommandAsJob` because that gets into Generator territory and we don't really need Generator here. But it is useful because it checks for npm and stuff right? Hm. Or maybe we should extract runNodeCommandAsJob out of Generator? Do we have a function runNodeCommand that doesn't run as a job? Should we create that one?